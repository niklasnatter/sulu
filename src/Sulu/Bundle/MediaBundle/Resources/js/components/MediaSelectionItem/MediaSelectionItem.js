// @flow
import React from 'react';
import MimeTypeIndicator from '../MimeTypeIndicator';
import mediaSelectionItemStyle from './mediaSelectionItem.scss';

type Props = {|
    mimeType: string,
    children: string,
    thumbnail?: string,
    thumbnailSize: number,
    thumbnailMargin: number,
|};

export default class MediaSelectionItem extends React.PureComponent<Props> {
    static defaultProps = {
        thumbnailMargin: 20,
        thumbnailSize: 25,
    };

    render() {
        const {
            mimeType,
            children,
            thumbnail,
            thumbnailSize,
            thumbnailMargin,
        } = this.props;

        const thumbnailStyles = {
            width: thumbnailSize,
            height: thumbnailSize,
            marginRight: thumbnailMargin,
        };

        return (
            <div className={mediaSelectionItemStyle.mediaSelectionItem}>
                <div style={thumbnailStyles}>
                    {thumbnail
                        ? <img
                            alt={thumbnail}
                            className={mediaSelectionItemStyle.thumbnailImage}
                            src={thumbnail}
                        />
                        : <MimeTypeIndicator
                            height={thumbnailSize}
                            iconSize={thumbnailSize / 1.6}
                            mimeType={mimeType}
                        />
                    }
                </div>
                {children}
            </div>
        );
    }
}
