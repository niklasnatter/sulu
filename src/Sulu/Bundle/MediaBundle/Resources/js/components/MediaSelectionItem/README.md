The `MediaSelectionItem` component displays a `thumbnail` or `mimeType` indicator beside a the given `children`. 
It is used by the `MediaSelection` and `SingleMediaSelection` to display the selected medias.

```javascript
const box = {
    "padding": "3px",
    "margin-bottom": "5px",
    "border": "1px solid gray",
};

<div>
    <div style={box}>
        <MediaSelectionItem
            mimeType="application/vnd.ms-excel"
            thumbnail="http://lorempixel.com/25/25"
        >
            media with thumbnail
        </MediaSelectionItem>
    </div>
    <div style={box}>
        <MediaSelectionItem
            mimeType="application/vnd.ms-excel"
        >
            media with mime type indicator
        </MediaSelectionItem>
    </div>
</div>
```
