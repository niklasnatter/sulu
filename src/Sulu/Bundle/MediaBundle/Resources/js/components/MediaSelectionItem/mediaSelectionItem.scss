$thumbnailSize: 25px;

.media-selection-item {
    display: flex;
    height: 100%;
    align-items: center;
}

.thumbnail-image {
    display: block;
    width: 100%;
    height: 100%;
}
