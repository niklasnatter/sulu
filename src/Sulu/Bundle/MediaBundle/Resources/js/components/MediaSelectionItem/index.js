// @flow
import MediaSelectionItem from './MediaSelectionItem';

export default MediaSelectionItem;
