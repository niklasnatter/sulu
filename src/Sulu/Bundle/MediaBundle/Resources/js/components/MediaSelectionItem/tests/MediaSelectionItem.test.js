// @flow
import {render} from 'enzyme';
import React from 'react';
import MediaSelectionItem from '../MediaSelectionItem';

test('Should render a MediaSelectionItem', () => {
    expect(render(
        <MediaSelectionItem
            mimeType="application/vnd.ms-excel"
            thumbnail="http://lorempixel.com/25/25"
        >
        test media
        </MediaSelectionItem>
    )).toMatchSnapshot();
});

test('Should render a MediaSelectionItem without thumbnail', () => {
    expect(render(
        <MediaSelectionItem
            mimeType="application/vnd.ms-excel"
        >
            test media
        </MediaSelectionItem>
    )).toMatchSnapshot();
});
