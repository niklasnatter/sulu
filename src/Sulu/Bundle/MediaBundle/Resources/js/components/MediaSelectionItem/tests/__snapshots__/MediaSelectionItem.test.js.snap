// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a MediaSelectionItem 1`] = `
<div
  class="mediaSelectionItem"
>
  <div
    class="thumbnail"
  >
    <img
      alt="http://lorempixel.com/25/25"
      src="http://lorempixel.com/25/25"
    />
  </div>
  test media
</div>
`;

exports[`Should render a MediaSelectionItem without thumbnail 1`] = `
<div
  class="mediaSelectionItem"
>
  <div
    class="thumbnail"
  >
    <div
      class="mimeTypeIndicator"
      style="color:#fff;font-size:16px;background-color:#00723a;height:25px"
    >
      <span
        aria-label="fa-file-excel-o"
        class="fa fa-file-excel-o"
      />
    </div>
  </div>
  test media
</div>
`;
